system/HTTP/ResponseTrait.php

```diff
@@ -759,39 +697,40 @@ public function getCookie(string $name = null, string $prefix = '')
 	 */
 	public function deleteCookie(string $name = '', string $domain = '', string $path = '/', string $prefix = '')
 	{
-		if (empty($name))
+		if ($name === '')
 		{
 			return $this;
 		}
 
-		if ($prefix === '' && $this->cookiePrefix !== '')
-		{
-			$prefix = $this->cookiePrefix;
-		}
+		$prefix = $prefix ?: Cookie::setDefaults()['prefix']; // to retain BC
```

Comment:
@paulbalandan Why do you need to call setter `setDefaults()`?

Reply:
An empty call to `setDefaults()` will just give the default cookie settings. That is like calling `getDefaults()`.

Reply:
Okay, but it is a tricky method.